envoy/config/extension_config_provider.h

```diff
@@ -62,6 +63,10 @@ class DynamicExtensionConfigProviderBase {
    * Applies the default configuration if one is set, otherwise does nothing.
    */
   virtual void applyDefaultConfiguration() PURE;
+  /**
+   * Return Network::ListenerFilterMatcherSharedPtr& the listener filter matcher.
+   */
+  virtual const Network::ListenerFilterMatcherSharedPtr& getListenerFilterMatcher() PURE;
```

Comment:
I think this needs to be ListenerFilterMatcherConstSharedPtr (i.e., the ListenerFilterMatcher is const, and not the pointer), no?

Reply:
This is to match with this interface definition, i.e, the pointer itself is const: https://github.com/envoyproxy/envoy/blob/6484dfe53bfdee2f4b94b09c31df58251aa6da0e/envoy/server/filter_config.h#L50

Reply:
I see, thanks!